python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -1098,6 +1098,113 @@ def test_composite_channel_multi_output(self, ray_start_regular_shared):
 
         compiled_dag.teardown()
 
+    def test_intra_process_channel_with_multi_readers(self, ray_start_regular_shared):
+        """
+        In this test, there are three 'echo' tasks on the same Ray actor.
+        The DAG will look like this:
+
+        Driver -> a.echo -> a.echo -> Driver
+                         |         |
+                         -> a.echo -
+
+        All communication between the driver and the actor will be done through remote
+        channels, i.e., shared memory channels. All communication between the actor
+        tasks will be conducted through local channels, i.e., IntraProcessChannel in
+        this case.
+        """
+        a = Actor.remote(0)
+        with InputNode() as inp:
+            dag = a.echo.bind(inp)
+            x = a.echo.bind(dag)
+            y = a.echo.bind(dag)
+            dag = MultiOutputNode([x, y])
+
+        compiled_dag = dag.experimental_compile()
+        ref = compiled_dag.execute(1)
+        assert ray.get(ref) == [1, 1]
+
+        ref = compiled_dag.execute(2)
+        assert ray.get(ref) == [2, 2]
+
+        ref = compiled_dag.execute(3)
+        assert ray.get(ref) == [3, 3]
+
+        compiled_dag.teardown()
+
+
+def test_simulate_pipeline_parallelism(ray_start_regular_shared):
+    """
+    This pattern simulates the case of pipeline parallelism training, where `w0_input`
+    reads data from the driver, and the fan-out tasks, `d00`, `d01`, and `d02`, use
+    `IntraProcessChannel` to read the data as the input for the forward pass.
+
+    Compared to reading data from shared memory channels for each forward pass, using
+    `IntraProcessChannel` may be more efficient because it avoids the overhead of
+    context switching and deserialization for each forward pass.
```

Comment:
Just to understand, what do you mean by "context switch" here?

Reply:
If multiple processes share the same page, it is more possible to have some synchronization operations to avoid race conditions which may cause more context switch between processes. I am OK to remove it if it is not clear.

Reply:
I removed "context switching" from the comment. https://github.com/ray-project/ray/pull/46431/commits/ff34fd84e7f473d4c5e3b24e677da8f041ffc078